src/hepattn/models/maskformer.py

```diff
@@ -126,11 +213,30 @@ def forward(self, inputs: dict[str, Tensor]) -> dict[str, Tensor]:
         x["query_embed"] = self.query_initial.expand(batch_size, -1, -1)
         x["query_valid"] = torch.full((batch_size, self.num_queries), True, device=x["query_embed"].device)
 
+        # Add positional encoding to the queries
```

Comment:
I'd extract this to a function `add_query_posenc` and call it right before (or inside) the decoder loop